src/edu/ucsb/nceas/metacat/dataone/quota/PortalQuotaService.java

```diff
@@ -0,0 +1,112 @@
+     */
+    private PortalQuotaService(ExecutorService executor, BookKeeperClient client) {
+        this.executor = executor;
+        this.client = client;
+    }
+    
+    /**
+     * Get a singleton class instance
+     * @return the PortalQuotaService instance
+     * @param executor  a thread executor service
+     * @param client  the client to connect the remote book keeper server
+     */
+    public static QuotaService getInstance(ExecutorService executor, BookKeeperClient client) {
+        if (service == null) {
+            synchronized (PortalQuotaService.class) {
+                if (service == null) {
+                    service = new PortalQuotaService(executor, client);
+                }
+            }
+        }
+        return service;
+    }
+    
+    @Override
+    public void enforce(String quotaSubject, Subject requestor, String instanceId, SystemMetadata sysmeta, String method) throws ServiceFailure, InvalidRequest, InsufficientResources, NotImplemented, NotFound, UnsupportedEncodingException {
+        logMetacat.debug("PortalQuotaService.enforce - checking both portal and storage quota types for the instance " + instanceId);
+        //this is to create a portal object. We should check both portal and storage quota
+        double portalQuantity = 1;
+        //double storageQuantity = sysmeta.getSize().doubleValue();
+        //int storageQuotaId = hasSpace(subscriber, requestor.getValue(), QuotaTypeDeterminer.STORAGE, storageQuantity, instanceId);
+        //report a portal usage in another thread
+        if (method != null && method.equals(QuotaServiceManager.CREATEMETHOD)) {
+            boolean checkSpace = true;
+            int portalQuotaId = checkQuota(checkSpace, quotaSubject, requestor.getValue(), QuotaTypeDeterminer.PORTAL, portalQuantity, instanceId);
+            createUsage(portalQuotaId, instanceId, portalQuantity);//report the usage in another thread
+        } else if (method != null && method.equals(QuotaServiceManager.ARCHIVEMETHOD)) {
+            if (isLastUnarchivedInChain(sysmeta.getIdentifier().getValue(), instanceId)) {
+                //take action only we are archiving the last object which hasn't been archived in the sid chain
+                boolean checkSpace = false;
+                int portalQuotaId = checkQuota(checkSpace, quotaSubject, requestor.getValue(), QuotaTypeDeterminer.PORTAL, portalQuantity, instanceId);
+                updateUsage(portalQuotaId, instanceId, portalQuantity);
+            } else {
+                logMetacat.debug("PortalQuotaService.enforce - Metacat is not archiving the last object which hasn't been archived in the series chain " + instanceId + ". It needs to do nothing for the portal quota");
+            }
+         } else if (method != null && method.equals(QuotaServiceManager.DELETEMETHOD)) {
+             if (isLastUnDletedInChain(sysmeta.getIdentifier().getValue(), instanceId)) {
```

Comment:
Fix this in other locations too:
```suggestion
             if (isLastUndeletedInChain(sysmeta.getIdentifier().getValue(), instanceId)) {
```

Reply:
See the commit b1a267ef091b34770080a7a67ffda970fe6ab2be.